src/components/Main/Modal/SettingModal/ThemeTab/EditTheme.vue

```diff
@@ -0,0 +1,151 @@
+<template>
+  <div :class="$style.content">
+    <!--
+    <form-button
+      label="エクスポート"
+      @click="onImportClick"
+      :disabled="isUpdated"
+      color="secondaly"
+      :class="$style.element"
+    />
+    -->
+    <form-button
+      label="インポート/エクスポート"
+      @click="onImportClick"
+      :disabled="isUpdated"
+      color="secondaly"
+      :class="$style.element"
+    />
+    <!--
+    <form-button
+      label="選択中のテーマで上書き"
+      @click="onUpdateClick"
+      color="error"
+      :class="$style.element"
+    />
+    -->
+  </div>
+  <template v-if="isUpdated">
+    <textarea-autosize
+      @input="val => (editedTheme = val.target.value)"
+      :value="editedTheme"
+      :class="$style.jsonarea"
+    />
+    <div :class="$style.updater">
+      <form-button
+        label="保存"
+        @click="onUpdateClick"
+        :disabled="isChanged"
+        color="primary"
+      />
+    </div>
+  </template>
+</template>
+
+<script lang="ts">
+import { defineComponent, computed, ref, PropType, SetupContext } from 'vue'
+import FormButton from '@/components/UI/FormButton.vue'
+import { Theme } from '@/types/theme'
+import { lightTheme } from '@/store/app/themeSettings/default'
+
+const hasKeyTheme = (theme: Theme): boolean => {
+  const keys = ['accent', 'background', 'text', 'ui']
+  const keysAccent = ['error', 'focus', 'notification', 'online', 'primary']
+  const keysBackground = ['primary', 'secondary', 'secondarySub', 'tertiary']
+  const keysUi = ['primary', 'secondary', 'tertiary']
+  const keysText = ['primary', 'secondary']
+  if (Object.keys(theme).sort().toString() !== keys.toString()) {
+    return false
+  }
+  if (Object.keys(theme.accent).sort().toString() !== keysAccent.toString()) {
+    return false
+  }
+  if (
+    Object.keys(theme.background).sort().toString() !==
+    keysBackground.toString()
+  ) {
+    return false
+  }
+  if (Object.keys(theme.ui).sort().toString() !== keysUi.toString()) {
+    return false
+  }
+  if (Object.keys(theme.text).sort().toString() !== keysText.toString()) {
+    return false
+  }
+  return true
+}
```

Comment:
これをしようとして型関連のエラー取れなくて諦めちゃったんですけどどうすればいいですか？
`Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'Theme'.
  No index signature with a parameter of type 'string' was found on type 'Theme'.`
というエラーで主に悩まされてるんですが……

Reply:
あ～たしかにそうですね

```ts
const hasKeyTheme = (theme: Theme): boolean => {
  return Object.keys(lightTheme).every(category => {
    const typedCategory = category as keyof Theme
    return Object.keys(lightTheme[typedCategory]).every(colorName => {
      const categoryColors = theme[typedCategory]
      if (!categoryColors) return false

      const typedColorName = colorName as keyof typeof categoryColors
      const color = categoryColors[typedColorName]
      return color && typeof color === 'string'
    })
  })
}
```
こうやったりすれば通るのですが、そもそもとして、これ受け取る型は`Theme`じゃなくて`any`なので下みたいにするのが良いと思います(最初から`Theme`型であることがわかってるならvalidationする必要がない)
```ts
const hasKeyTheme = (theme: any): boolean => {
  return (Object.keys(lightTheme) as Array<keyof Theme>).every(category =>
    Object.keys(lightTheme[category]).every(
      colorName =>
        theme[category] &&
        theme[category][colorName] &&
        typeof theme[category][colorName] === 'string'
    )
  )
}
```

Reply:
anyを使うとESlintがwarning出してくるの、どうすればいいですか？

Reply:
`eslint-disable-next-line ～`って書いてその行だけ無効化してください 🙏 
VSCodeなら波線のところで 💡 を押すとそれを挿入できるやつボタンがあるはずです(`eslint`の拡張機能入れないと出ないかも)